MainWindow.cpp

```diff
@@ -469,8 +469,35 @@ void MainWindow::on_actionOpen_triggered() {
 }
 
 void MainWindow::on_actionSave_triggered() {
-  const QString &filter = "EGM (*.egm)";
-  const QString &fileName = QFileDialog::getSaveFileName(this, tr("Save Project"), "", filter);
+  // map useful to quickly fetch extension from the selected filter in QDialog (because Qt doesn't give a better way)
+  std::unordered_map<QString, QString> extensionMap;
+  extensionMap["EGM (*.egm)"] = ".egm";
+
+  // lambda to create a QString of all filters separated by ";;" except last. The generated QString is required
+  // as an argument in getSaveFileName
+  auto getFilters = [&extensionMap]() -> QString {
+    assert(static_cast<int>(extensionMap.size()) > 0);
+
+    auto itr = extensionMap.begin();
+    QString f = itr->first;
+    ++itr;
+
+    for(; itr != extensionMap.end(); ++itr){
+      f += ";;" + itr->first;
+    }
+    return f;
+  };

```

Comment:
If you use `QHash` instead of `std::unordered_map` you can just write:
```c++
const auto filters = extensionMap.keys().join(QStringLiteral(";;"));
```
I'm sure something is possible with STL, but in general Qt containers provide a little more convenience. :-)

However, why do we do this given that we support only one file format anyway? Are additional file formats planned for the Save function?

Reply:
I guess I liked qt way of doing that more, didn't know that, chages on way. I'm not sure about supporting extensions in future, just thought a little ahead of time, that maybe someone adds another extension.

Reply:
Additional formats are possible but not planned as of yet. Someone way want to do gmx export later.